qcodes/dataset/data_set_protocol.py

```diff
@@ -427,6 +430,7 @@ def _export_data(
         # Set defaults to values in config if the value was not set
         # (defaults to None)
         path = path if path is not None else get_data_export_path()
+        os.makedirs(path, exist_ok=True)
```

Comment:
as you switched to pathlib here you could consider using Path.mkdirs, and we could consider also creating the parents.